### ISSUE TITLE
Keras and Larq summaries show different results on FP models

### ISSUE THREAD

**Issue by mervess** [larq/larq#716]
When I use _Keras_'s `m.summary()` and _Larq_'s `lq.models.summary(m)` for the same full-precision (FP) model, I get different results.

I expect to see **Params** to remain the same. However, what _Keras_ shows as the **Trainable params**, _lq_ shows it as the **Total params** count. Especially Batch Normalization layers are counted as the half of the _Keras_'s shown value. Is there any reason behind it?

Thanks in advance.

**Comment by lgeiger** [comment]
@mervess Very sorry for the late reply.

> Especially Batch Normalization layers are counted as the half of the Keras's shown value. Is there any reason behind it?

We decided to add a special case to Batch Normalization layers in #454. This is due to the fact the Keras Batch Normalization layer usually uses 4 floating point variables (`moving_mean`, `moving_variance`, `gamma` and `beta`). However during inference these values will always be fused into just 2 which is why we deviate from the Keras model summary here in the hope to provide a more "correct" estimate of the parameters of the final inference graph since for binary networks this difference can become quite significant.

As mentioned in #454 this is a bit of a slippery slope since we can't anticipate all conversion-time optimizations in the profile (in many cases the Batch Normalization layer can even be completely fused into the preceding convolution or fully connected layer which would make it free). In general the model profile is a best effort estimate of the number of parameters and number of MACs. The true computation will always be determined by the inference framework and the optimisations that can be done on the inference graph itself.

Please let me know if this clears things up.

**Comment by mervess** [comment]
I see, that makes sense.
Indeed, thanks @lgeiger.
